src/glualintProvider.ts

```diff
@@ -32,6 +33,28 @@ export default class GLuaLintingProvider implements vscode.Disposable {
 
         const config = vscode.workspace.getConfiguration('glualint');
 
+        // First all files
+        const allFiles = config.get<boolean>('allFiles');
+        if (allFiles) {
+            // HACK: bypasses activeLanguages setting!
+            const allFiles = await vscode.workspace.findFiles('**/*.lua');
```

Comment:
Something I've only now realized is that this might not work with virtual workspaces. I think I already noted somewhere that only single file linting works in this case, but might want to make sure the error checking below does not blow up incase you attempt to do it anyways.

Reply:
It didn't blow up with virtual workspaces in my testing, but I went ahead and handled it better for this use case, so it will now fallback to the previous method of linting all files separately and actually work as a result.

I have also applied a concurrency limit to `lintUri` to ensure we only spawn 50 processes a time so the PC is not too swamped for large projects.